tests/builder/test_base.py

```diff
@@ -122,11 +122,7 @@ def test_run_exception(builder):
     if sys.version_info >= (3, 6):
         builder.build.assert_called()
     # bulid should be called 3 times for retry setting.
-    calls = [
-        mock.call({'name': 'a'}),
-        mock.call({'name': 'a'}),
-        mock.call({'name': 'a'}),
-    ]
+    calls = [mock.call({'name': 'a'}, retry=3)] * 3
```

Comment:
I think adding below cases can be better. This line shows called 3 times. but should be called 4 times.

If `builder.run(mock_work, retry=3)`, => builder.build should be called 4 times (not 3 times, not 2 times.).
If `builder.run(mock_work, retry=0)`, => builder.build should be called 1 time (not 2 times).

(Sorry my previous retry specification was my mistake. I called `builder.build` total 3 times for retry=3).

But interestingly `builder.build.assert_has_calls(calls)` is passed for current 3 times calls.
So, maybe `assert_has_calls` can not test how many times the method is called with same argument.